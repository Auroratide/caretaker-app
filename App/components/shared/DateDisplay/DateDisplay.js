import React from 'react';
import { Text, View } from 'react-native';
import styles from './style';

const dayOfTheWeek = date => ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'][date.getDay()];

export default function DateDisplay( { date } ) {
  return <View style={styles.dateDisplay}>
    <Text style={styles.dateDisplayDate}>{date.getDate()}</Text>
    <Text style={styles.dateDisplayDay}>{dayOfTheWeek(date)}</Text>
  </View>;
}
