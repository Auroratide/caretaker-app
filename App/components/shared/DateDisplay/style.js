import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  dateDisplay: {
    width: 56,
    padding: 4
  },
  dateDisplayDate: {
    fontSize: 24
  },
  dateDisplayDay: {}
});

export default styles;
