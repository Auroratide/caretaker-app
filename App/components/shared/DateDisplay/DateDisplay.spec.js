import React from 'react';
import renderer from 'react-test-renderer';
import DateDisplay from './DateDisplay';

describe('<DateDisplay />', () => {
  const Jul_10_2017 = new Date(1499662800000);

  let props;
  const dateDisplay = () => renderer.create(<DateDisplay {...props} />);

  beforeEach(() => {
    props = {
      date: undefined
    };
  });

  it('should render correctly', () => {
    props.date = Jul_10_2017;
    expect(dateDisplay()).toMatchSnapshot();
  });

});
