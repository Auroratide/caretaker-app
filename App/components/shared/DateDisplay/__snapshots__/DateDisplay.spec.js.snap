// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateDisplay /> should render correctly 1`] = `
<View
  style={
    Object {
      "padding": 4,
      "width": 56,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 24,
      }
    }
  >
    10
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={Object {}}
  >
    Mon
  </Text>
</View>
`;
